paderbox/array/kernel.py

```diff
@@ -0,0 +1,212 @@
+import typing
+import numpy as np
+import paderbox as pb
+
+
+__all__ = [
+    'np_kernel',
+    'ai_dilate',
+    'ai_erode',
+    'max_kernel',
+    'min_kernel',
+    'mean_kernel',
+    'median_kernel',
+]
+
+
+def _plot(a):
+    """
+    >>> _plot([0, 1, 2, 3, 4, 5, 6, 7, 8])
+    ' ▁▂▃▄▅▆▇█'
+    """
+    a = np.array(a)
+    values = np.array(list(' ▁▂▃▄▅▆▇█'))
+    m = np.amax(a)
+    if m == 0:
+        m = 1
+    a = np.round(a * ((len(values)-1) / m)).astype(int)
+    print(repr(''.join(values[a])))
+
+
+def np_kernel1d(
+        x,
+        kernel_size,  # an odd number
+        *,
+        kernel: callable,  # e.g. np.mean, np.amax, np.amin
+        padding_mode: 'typing.Literal["edge"]' = 'edge',
+        pad_position: 'typing.Literal["pre", "post", None]' = 'pre',
+):
```

Comment:
Does it make sense to add an `axis` argument? You could use `np.moveaxis` to move it to the back, apply the same code you already have, and then move the axis back to its original position.

Reply:
The axis argument would not have an effect if ArrayIntervals are used, right? I thunk that's not a behavior we want.

Reply:
`ArrayInterval`s are always 1D, so forcing `axis=0` is reasonable. This function could also be useful outside of this module where `ArrayInterval`s are not needed.

Reply:
Done. `np_kernel1d` has now an axis argument, but I kept the thin wrappers with the minimum number of arguments, so they can work with different backends.

Reply:
We can add the axis argument to the wrappers, once we have a use case.